### DISCUSSION TITLE
Optimising Motion Planning Parameters for more deterministic behaviour.

### DISCUSSION THREAD

**Discussion by D-1shu** [NVlabs/curobo#215] [Software Q&A]
Hi developers,

cuRobo's overall concept and execution is very impressive. In my experiments with a custom dual-arm setup, it successfully generated smooth trajectories. To further enhance motion planning for my specific use case, I have a few questions:

#### Integration with Graph Planners: 
I'm interested in exploring how cuRobo can potentially leverage graph-based motion planning?
#### Default Behavior
Is graph-based planning enabled by default in cuRobo's motion planning pipeline? Or is it triggered under 
specific conditions (e.g., highly complex environments)?
#### Customization Options
Does cuRobo offer the flexibility to disable it if my use case doesn't necessitate it?

### Manually Provided IK Seeds:

For certain scenarios in my dual-arm setup, I might have a good initial guess for the desired Inverse Kinematics (IK) solution. Can cuRobo incorporate manually provided IK seeds as starting points for the trajectory optimization process?

### Trajectory Sample Injection:

As an additional strategy for refinement, I'd like to explore the possibility of injecting some manually generated trajectory samples as input to the motion planner. Would cuRobo be able to leverage these samples to guide and optimize the final trajectory?

**Goal: Deterministic and Repeatable Behavior**
My overall objective is to achieve more deterministic and repeatable motion planning behavior, particularly when the start and goal poses for the robot arms are repeated. I believe that the strategies mentioned above (IK seeds, and trajectory samples) have the potential to contribute to this goal.

I appreciate any insights or guidance you can provide on these questions. Thank you for your time and for developing this excellent library!

**Comment by scharalambous-sony** [comment]
Graph-based motion planning is used to generate the seeds for the TO. Whether this is used is determined by _enable_graph_attempt_
_enable_graph_attempt_ only uses the graph motion planner if it failed to find a solution within the attempts specified by this parameter

About manually providing seeds, I'd also be interested to know if it's possible:)
